src/rust/ensogl/lib/components/src/toggle_button.rs

```diff
@@ -64,6 +64,94 @@ impl<Shape:ColorableShape+'static> Model<Shape> {
 
 
 
+// ===================
+// === ButtonState ===
+// ===================
+
+/// A state a button can be in.
+#[derive(Clone,Copy,Debug,Default)]
+#[allow(missing_docs)]
+pub struct ButtonState {
+    pub visible : bool,
+    pub enabled : bool,
+    pub hovered : bool,
+    pub pressed : bool,
```

Comment:
It might be good to explain the states `enabled` and `pressed` as they can have overlapping meaning. I assume one of them is the `toggle state` and one of them is the `can-be-toggled state`, but the names are not 100% self-explaining.